Code/Framework/AzCore/Tests/Debug/PerformanceCollectorTests.cpp

```diff
@@ -0,0 +1,215 @@
+        // parameter. Each json object will contain statistical summary of a batch made of 10 frames.
+
+        const AZStd::string LogCategory("PerformanceCollectorTest");
+        constexpr AZStd::string_view PerfParam1("param1");
+        constexpr AZStd::string_view PerfParam2("param2");
+        constexpr AZStd::string_view PerfParam3("param3");
+
+        // Define control arguments that will define how much data to collect.
+        // In real life these parameters are mirrored by CVARs.
+        const AZ::u32 numberOfCaptureBatches = 3;
+        const AZ::Debug::PerformanceCollector::DataLogType dataLogType = AZ::Debug::PerformanceCollector::DataLogType::LogStatistics;
+        const AZ::u32 waitTimePerCaptureBatch = 0; //0 seconds
+        const AZ::u32 frameCountPerCaptureBatch = 10;
+
+        AZ::u32 pendingBatchCount = numberOfCaptureBatches;
+        auto onCompleteCallback = [&](AZ::u32 batchCount) {
+            pendingBatchCount = batchCount;
+        };
+        AZStd::vector<AZStd::string_view> paramList = { PerfParam1, PerfParam2, PerfParam3 };
+        AZ::Debug::PerformanceCollector performanceCollector(LogCategory, paramList, onCompleteCallback);
+        performanceCollector.UpdateDataLogType(dataLogType);
+        performanceCollector.UpdateFrameCountPerCaptureBatch(frameCountPerCaptureBatch);
+        performanceCollector.UpdateWaitTimeBeforeEachBatch(AZStd::chrono::seconds(waitTimePerCaptureBatch));
+        performanceCollector.UpdateNumberOfCaptureBatches(numberOfCaptureBatches);
+
+        // We will loop more than is required to simulate OnSystemTick, but the amount of collected data should
+        // be limited to the specified control parameters.
+        const AZ::u32 EXCESS_FRAME_LOOP = 5;
+        for (AZ::u32 frameLoop = 0; frameLoop < ((numberOfCaptureBatches*frameCountPerCaptureBatch) + EXCESS_FRAME_LOOP); frameLoop++)
+        {
+            performanceCollector.FrameTick();
+            if (!performanceCollector.IsWaitingBeforeCapture())
+            {
+                performanceCollector.RecordPeriodicEvent(PerfParam1);
+            }
+            {
+                AZ::Debug::ScopeDuration param1Scope(&performanceCollector, PerfParam2);
+            }
+            {
+                AZ::Debug::ScopeDuration param1Scope(&performanceCollector, PerfParam3);
+            }
+        }
+
+        auto stringWithOutputData = performanceCollector.GetOutputDataBuffer();
+        EXPECT_FALSE(stringWithOutputData.empty());
+        EXPECT_EQ(pendingBatchCount, 0);
+
+        rapidjson::Document jsonDoc;
+        jsonDoc.Parse(stringWithOutputData.c_str());
+        ASSERT_TRUE(!jsonDoc.HasParseError());
+
+        // The root is an array of dictionaries.
+        ASSERT_TRUE(jsonDoc.IsArray());
+        // We have 3 paramaters and 3 capture batches. So we expect 9 arrays.
+        auto expectedArraySize = paramList.size() * numberOfCaptureBatches;
+        ASSERT_TRUE(jsonDoc.Size() == expectedArraySize);
```

Comment:
This should use ASSERT_EQ
```suggestion
        ASSERT_EQ(expectedArraySize, jsonDoc.Size());
```

Reply:
Fixed.